src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs

```diff
@@ -849,7 +850,14 @@ private bool ShouldTryToReconnect(Exception ex)
                 case AuthorizationException _:
                 case AllTopologyNodesDownException _:
                 case SubscriberErrorException _:
-                case RavenException _:
+                    _processingCts.Cancel();
+                    return false;
+                case RavenException re:
+                    if (re.InnerException is HttpRequestException or TimeoutException)
+                    {
+                        goto default;
+                    }
+
```

Comment:
I wonder if we want to `    OnUnexpectedSubscriptionError?.Invoke(ex);` here as well?

Reply:
I do not think so, because this is not unexpected?